enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,401 @@
+   verbs: ["create"]
+```
+3. As a DEV/QA team member, I can `create` a managedCluster, and add label `cluster.open-cluster-management.io/clusterset:devset/qaset` to the managedCluster.
+4. As a DEV/QA team member, I could `bind` the managedClusterSet `devset/qaset` to my namespace and run applications in the managedClusterSet.
+
+## Test Plan
+- Unit tests will cover the functionality of the controllers.
+- Unit tests will cover the new APIs.
+- And e2e tests will cover the following cases.
+ - Create/update/delete managedClusterSet 
+ - Create/update/delete managedClusters labels
+ - Update managedClusterSet spec `clusterSelector` field
+
+## Migration
+Currently, managedClusterSet has three consumers: [placement](https://github.com/open-cluster-management-io/placement), [multicloud-operators-foundation](https://github.com/stolostron/multicloud-operators-foundation), [submariner-addon](https://github.com/stolostron/submariner-addon)
+
+- `multicloud-operators-foundation` uses managedClusterSet for resource group purposes. And it should only care about the exclusive managedClusterSet. 
+- `submariner-addon` uses managedClusterSet to group clusters based on the networks. And in different managedClusterSet, the clusters should be exclusive.
+- `placement` select all managedClusters in each managedClusterSet.
+
+So we could finish the migration by four steps, and step 1 and step 2 will be finished in OCM 0.7.0. and step 3 and step 4 will be finished in OCM 0.8.0
+
+1. [Implement in OCM 0.7.0]Update the managedClusterSet API which only includes an exclusive way to select target managedClusters.
+
+```go
+type ManagedClusterSetSpec struct {
+    // Selector represents a selector of ManagedClusters by labels and names.
+    ClusterSelector ManagedClusterSelector  `json:"clusterSelector"`
+}
+
+type ManagedClusterSelector struct{
+    // "" means to use the current mechanism of matching label <cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>.
+    // (future) "LabelSelector" means to use the LabelSelector to select target managedClusters
+    // "ClusterLabel" means to use a particular cluster label. It is guaranteed that clustersets with same label key are exclusive with each others
+    // +optional
+    SelectorType SelectorType `json:"selectorType"`
+
+    // ClusterLabel defines one label which clusterset could use to select target managedClusters. In this way, we will:
+    // 1. Guarantee clustersets with same label key are exclusive
+    // 2. Enable additional permission check when cluster joining/leaving a clusterset (the label key should start with the reserved prefix "cluster.open-cluster-management.io/" and "info.open-cluster-management.io/");
+    ClusterLabel *ClusterLabel `json:"clusterLabel"`
+}
+
+type SelectorType string
+
+const (
+    ClusterLabel        SelectorType = "ClusterLabel"
+)
+
+//ClusterLabel defines one cluster label
+type ClusterLabel struct {
+	  Key string `json:"key"`
```

Comment:
key by default is cluster.open-cluster-management.io/clusterset and can only be cluster.open-cluster-management.io/